### PR TITLE
experiment with using github api to determine xtext version

### PR DESCRIPTION
**Pull request by cdietrich**
experiment with using github api to determine xtext version
Fixes https://github.com/eclipse/xtext/issues/2036

### CODE REVIEW

File: releng/jenkins/check-versions/Jenkinsfile [hunk 1]
```diff
@@ -215,25 +224,30 @@ def getXtextGradlePluginVersion (branch) {
  * @param repository (Optional) Xtext repository name on GH
  */
 def getVersionFromGradleVersions (id,branch='master',repository='xtext-lib') {
-  return sh (script: "curl -sS https://raw.githubusercontent.com/eclipse/${repository}/${branch}/gradle/versions.gradle |grep -Po \"${id}[^\\d]*\\K[^']*\"", returnStdout: true).trim()
+  env.BRANCH_REF="${branch}"
```
Comment (cdietrich) [inline releng/jenkins/check-versions/Jenkinsfile:227]: if there a way to avoid such thing?

File: releng/jenkins/check-versions/Jenkinsfile [hunk 2]
```diff
@@ -243,5 +257,9 @@ def getXtextBootstrapVersion (branch) {
  */
 
 def getVersionFromBOM (id, branch='master') {
-  return sh (script: "curl -sS https://raw.githubusercontent.com/eclipse/xtext-lib/${branch}/org.eclipse.xtext.dev-bom/build.gradle |grep -Po 'api \\\"${id}:\\K[^\"]*'", returnStdout: true).trim()
+  env.BRANCH_REF="${branch}"
+  sh (script: 'curl -sSL -H "Cookie: logged_in=no" -H "Authorization: token $GITHUB_API_TOKEN" -H "Content-Type: text/plain; charset=utf-8" https://api.github.com/repos/eclipse/xtext-lib/contents/org.eclipse.xtext.dev-bom/build.gradle?ref=$BRANCH_REF | jq -r ".content" | base64 -d > av.txt')
```
Comment (cdietrich) [inline releng/jenkins/check-versions/Jenkinsfile:261]: i dont get this running without using a file.
either the interpolation wont work or the grep

File: releng/jenkins/check-versions/Jenkinsfile (hunk 1 above)
Comment (nprediger) [reply releng/jenkins/check-versions/Jenkinsfile:227]: there's only a couple of ways to get values into env vars. you could use the environment {} block and call a function to set its values

File: releng/jenkins/check-versions/Jenkinsfile (hunk 2 above)
Comment (nprediger) [reply releng/jenkins/check-versions/Jenkinsfile:261]: i think using a file is not that bad. even less strain on the rate limit

**Review by nprediger — APPROVED** [review]

File: releng/jenkins/check-versions/Jenkinsfile (hunk 1 above)
Comment (cdietrich) [reply releng/jenkins/check-versions/Jenkinsfile:227]: no i mean, is there a way to compile string interpolaton with non string interpolation

Comment (cdietrich) [reply releng/jenkins/check-versions/Jenkinsfile:227]: i just nedd to create the ENV crap so that it gets injected into the single quoted strings

**Pull request merged**